File: app/src/main/java/com/hyphenate/graduationdesignchat/Md5Utils.java
package com.hyphenate.graduationdesignchat;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/**
 * Created by Scus on 2018/3/16.
 */

public class Md5Utils {

    public static String md5Password(String password) {
        StringBuffer stringBuffer = new StringBuffer();

        try {
            MessageDigest messageDigest = MessageDigest.getInstance("MD5");
            byte[] digest = messageDigest.digest(password.getBytes());
            //每个byte做一个与运算
            for (Byte b : digest) {
                int i = b & 0xff;
                String hexString = Integer.toHexString(i);
                if (hexString.length() == 1) {
                    stringBuffer.append("0");
                }
                stringBuffer.append(hexString);
            }
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        return stringBuffer.toString();
    }
}

File: app/src/main/java/com/hyphenate/graduationdesignchat/MD5Utils.kt
package com.hyphenate.graduationdesignchat

import java.security.MessageDigest

/**
 * Created by Scus on 2018/3/16.
 */
class MD5Utils {
    companion object {
        var stringBuffer: StringBuffer = StringBuffer()
        fun md5Password(password: String): String {
            val instance = MessageDigest.getInstance("MD5")
            val digest = instance.digest(password.toByteArray())
            for (b in digest) {
                val i = b.toInt() and 0xff
                var hexString = Integer.toHexString(i)
                if (hexString.length < 2) {
                    hexString = "0" + hexString
                }
                stringBuffer.append(hexString)
            }
            return stringBuffer.toString()
        }
    }
}